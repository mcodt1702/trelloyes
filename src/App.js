import React, { Component } from "react";
import List from "./List";
import "./App.css";

class App extends Component {
  state = { ...this.props.store };

  handleDeleteCard = (listId, cardId) => {
    this.setState({
      lists: this.state.lists.map((list) => {
        list.cardIds =
          list.id === listId
            ? list.cardIds.filter((cId) => cId !== cardId)
            : list.cardIds;
        return list;
      }),
    });
  };




  
    newRandomCard = () => {
      const id = Math.random().toString(36).substring(2, 4)
        + Math.random().toString(36).substring(2, 4);
      return {
        id,
        title: `Random Card ${id}`,
        content: 'lorem ipsum',
      }
    

 


  }

  render() {
    const { lists, allCards } = this.state;
    return (
      <main className="App">
        <header className="App-header">
          <h1>Trelloyes!</h1>
        </header>
        <div className="App-list">
          {lists.map((list, i) => (
            <List
              key={list.id}
              header={list.header}
              id={list.id}
              cards={list.cardIds.map((id) => allCards[id])}
              handleDeleteCard={this.handleDeleteCard}
              newRandomCard={this.newRandomCard}
            />
          ))}
        </div>
      </main>
    );
  }
}

export default App;
