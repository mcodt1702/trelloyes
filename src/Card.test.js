import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Card from './Card';
import List from './List';
import renderer from 'react-test-renderer';




describe('Card Componen Suite', () =>{
    it ('renders without crashing', ()=>{

        const div = document.createElement('div');
        ReactDOM.render(<App/>, div);

        ReactDOM.unmountComponentAtNode(div);
            }   
    )

    it ('renders the UI as expected', () => {
        const tree = renderer
        .create(<Card
            key= {4}
            title= 'any'
            content= 'any content'
          />)
        .toJSON();
        expect(tree).toMatchSnapshot()


    })





})