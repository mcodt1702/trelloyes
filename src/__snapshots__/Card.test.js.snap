// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Componen Suite renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    any
  </h3>
  <p>
    any content
  </p>
</div>
`;
